### PR TITLE
Fix Usage grouping in the "Find" window

### LINKED ISSUE #2672: Group by usage type in Find Usages results with wrong grouping
Without grouping:

image

With grouping (wrong):

image

### PR DESCRIPTION
**Pull request by kumbayo**
Currently the "Group by usage type" option in the Find window is broken.
Each result is placed in its own group.
image

Fix this problem
image

This also includes the following smaller changes:
- Support `macro call` usage.
- Change `trait ref` to `trait reference` to it better align with the `type reference` usage type.

This fixes #2672

### CODE REVIEW

**Comment by Undin** [comment]
Looks great!
Let's add tests for Usage View. You can find simple example [here](https://github.com/JetBrains/intellij-community/blob/master/java/java-tests/testSrc/com/intellij/java/psi/usages/JavaUsageViewTreeTest.java)
Unfortunately, `com.intellij.testFramework.fixtures.CodeInsightTestFixture#getUsageViewTreeTextRepresentation` method is available only in 183 platform. But we can create such test only for 183 platfrom for now.
I've improved platform dependent source layout in #3056 (rebase onto it please) for such tests creation. Now you just need to create your test class in `183/kotlin/test` directory. Change `platformVersion` property to `183` in `gradle.properties` file to import the project with 183 platform.

**Comment by kumbayo** [comment]
@Undin I added the test and rebased on master.

File: src/183/test/kotlin/org/rust/ide/search/RsUsageViewTreeTest.kt [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    """)
+
+    private fun doTestByText(@Language("Rust") code: String, representation: String) {
+        InlineFile(code)
+        val source = findElementInEditor<RsNamedElement>()
+
+        val searchResult = ReferencesSearch.search(source).findAll()
+        val usageInfos = searchResult.map { UsageInfo(it) }
+
+        val textRepresentation = myFixture.getUsageViewTreeTextRepresentation(usageInfos)
```
[the remark this answers is not in the document]
Comment (kumbayo) [reply src/183/test/kotlin/org/rust/ide/search/RsUsageViewTreeTest.kt]: Done (the second suggestion)

**Comment by Undin** [comment]
Excellent!
Thanks a lot!
bors r+

**Pull request merged**